src/other/utils.jl

```diff
@@ -221,16 +221,22 @@ end
 Equivalent to `Threads.@spawn` if `threads === true`,
 otherwise run `expr` and return a `Task` that returns its value.
 """
-macro spawn_or_run_task(threads, expr)
-    letargs = Base._lift_one_interp!(expr)
+macro spawn_or_run_task(threads, ex)
+    letargs = Base._lift_one_interp!(ex)
 
-    thunk = esc(:(()->($expr)))
+    thunk = :(()->($(esc(ex))))
+    @static if VERSION >= v"1.10.0-DEV"
+        Base.replace_linenums!(thunk, __source__)
```

Comment:
I understand we do not need `thunk = Base.replace_linenums!(thunk, __source__)`?

Reply:
I'm not sure what this does but I assume this improves line number in stacktraces?

Reply:
No - I meant the `thunk =` part, i.e. does `Base.replace_linenums!` guarantee to update `thunk` in place?

Reply:
Ah good point. Yes, it always returns `thunk`.